R/uniDimFrequentistOutput.R

```diff
@@ -0,0 +1,183 @@
+
+
+.frequentistScaleTable <- function(jaspResults, model, options) {
+
+  if (!is.null(.getStateContainerF(jaspResults)[["scaleTable"]]$object))
+      return()
+
+  scaleTable <- createJaspTable(gettext("Frequentist Scale Reliability Statistics"))
+  scaleTable$dependOn(options = c("omegaScale", "alphaScale", "lambda2Scale", "lambda6Scale", "glbScale",
+                                  "averageInterItemCor", "meanScale", "sdScale", "meanMethod", "sdMethod"))
+  scaleTable$addColumnInfo(name = "estimate", title = gettext("Estimate"), type = "string")
+
+  if (options[["intervalOn"]]) {
+    intervalLow <- gettextf("%s%% CI",
+                            format(100*options[["confidenceIntervalValue"]], digits = 3, drop0trailing = TRUE))
+    intervalUp <- gettextf("%s%% CI",
+                           format(100*options[["confidenceIntervalValue"]], digits = 3, drop0trailing = TRUE))
+    intervalLow <- gettextf("%s lower bound", intervalLow)
+    intervalUp <- gettextf("%s upper bound", intervalUp)
+    allData <- data.frame(estimate = c(gettext("Point estimate"), intervalLow, intervalUp))
+  } else {
+    allData <- data.frame(estimate = c(gettext("Point estimate")))
+  }
+
+  derivedOptions <- model[["derivedOptions"]]
+  opts     <- derivedOptions[["namesEstimators"]][["tables"]]
+  selected <- derivedOptions[["selectedEstimators"]]
+  idxSelected <- which(selected)
+
+  # if no coefficients selected:
+  if (!is.null(model[["empty"]])) {
+    scaleTable$setData(allData)
+    nvar <- length(options[["variables"]])
+    if (sum(selected) > 0L && nvar < 3) {
+      for (i in idxSelected) {
```

Comment:
is this check necessary?
when `sum(selected) > 0L` would be `FALSE`, then `for (i in idxSelected)` would never be entered. I'm not 100% sure why `nvar` is checked either?

Reply:
you are right